### PR TITLE
Add (almost) full support for long arithmetic

### PR DESCRIPTION
**Pull request by swenson**
This fixes most of https://github.com/pybee/batavia/issues/223.

Things left (in a future PR): logical functions, more tests.

### CODE REVIEW

**Comment by swenson** [comment]
Whew! Ready for review now.

The existing tests were really thorough, and I ended up having to go ahead and convert almost everything over to supporting long arithmetic to get the tests to pass. :)

Let me know if you want me to squash all of this.

**Comment by glasnt** [comment]
@swenson 🎉 

I can squash it all when I merge, it's all good :)

**Comment by swenson** [comment]
Sounds good to me!

**Comment by glasnt** [comment]
So I *think* this looks right (and the robot agrees), but given it's touching a bunch of marshelling code, I'm going to defer to the Bee-nevolent @freakboy3742 for mergification

**Comment by swenson** [comment]
The more the merrier!

File: batavia/core/builtins.js [hunk 1]
```diff
@@ -1143,10 +1149,10 @@ batavia.builtins.sum = function(args, kwargs) {
         throw new batavia.builtins.TypeError('sum() expected at most 2 argument, got ' + args.length);
     }
 
-    var total = 0;
+    var total = new batavia.types.Int(0);
```
Comment (freakboy3742) [inline batavia/core/builtins.js]: It's a *really* minor optimisation; but is this initialisation required?

**Comment by freakboy3742** [comment]
LGTM. Lots of good "attention to detail" stuff that has slipped in where we haven't been clear about the use of native integers or batavia.types.Int. Great stuff!

I picked up one *really* minor optimisation; there might also be room to normalize the naming of the `r_long` et al methods so that they are consistently named. Other than that, it all makes sense to me.

File: batavia/core/builtins.js (hunk 1 above)
Comment (swenson) [inline batavia/core/builtins.js]: Yes, it turns out it is (for `sum([])`, but I didn't do it correctly. Added a test for this and fixed it in another commit.

**Comment by swenson** [comment]
I added a fix for the minor optimisation -- let me know if you also want me to redo some of the `r_long` etc. names.

Thanks!

**Comment by glasnt** [comment]
Merged and squished! Thank you @swenson! ✨

**Comment by swenson** [comment]
Thank you so much!

**Pull request merged**